Create role-lead unresponsiveness process (#2322)
kubernetes/sig-release: release-team/release-team-removal.md

> # Resignation

Reviewer: ```suggestion
# Release Team Removal Process

The release team is reassembled with a new team for each cycle. All team members are automatically removed if they do not apply and are selected for the next release cycle. This process works well, but from time to time a team member may leave the release team in the middle of the release cycle or the release team leaders may decide to part ways early. This document describes the process for this eventuality.

## Resignation
```

Reviewer: lets add some intro to the document.

Reviewer 2: I agree- I think an intro would be helpful. Maybe rephrase slights: 

```
All team members are automatically removed if they do not re-apply and are selected for the next release cycle. While this process generally functions smoothly, there are instances when a team member might leave mid-cycle or the release team leaders may decide to part ways early. 
```

> Voluntary resignation is outlined in the [shadow handbook](https://github.com/leonardpahlke/sig-release/blob/master/release-team/shadows.md). Role lead should discuss with Release Lead and Emiritus Advisor to find a replacement if they would like to step down.

Reviewer: ```suggestion
Voluntary resignation is outlined in the [shadow handbook](https://github.com/leonardpahlke/sig-release/blob/master/release-team/shadows.md). Role lead should discuss with Release Lead and Emeritus Advisor to find a replacement if they would like to step down.
```

> # Removal Process

Reviewer: ```suggestion
```

> Remaining on the team without contributing creates an unhealthy environment for the rest of the team, and negatively affect the outcome of the release. We strive to be inclusive and empathetic to individuals circumstances. 

Reviewer: ```suggestion
Remaining on the team without contributing creates an unhealthy environment for the rest of the team, and negatively affects the outcome of the release. We strive to be inclusive and empathetic to individuals circumstances.
```

Reviewer 3: At the end of the cycle, the leads of each team could confirm a list of the shadows abandon their duties and be removal from the github release team (https://github.com/kubernetes/sig-release/blob/master/releases/release-1.xx/release-team.md)

> Remaining on the team without contributing creates an unhealthy environment for the rest of the team, and negatively affect the outcome of the release. We strive to be inclusive and empathetic to individuals circumstances. 

Reviewer 2: ```suggestion
Remaining on the team without contributing creates an unhealthy environment for the rest of the team and negatively affect the outcome of the release. We strive to be inclusive and empathetic to individuals' circumstances. 
```

The following outline is not meant to be definitive or punitive but rather outlines general guidance for leads on the process.  
## Shadow Removal Guideline

Reasons for removal could be including but not limited to:

> - Failure to attend shadow orientation hosted by EA (we host 2-3 sessions in different timezone)

Reviewer: ```suggestion
- Failure to attend shadow orientation hosted by EA (we host 2–3 sessions in different timezones)
```

> - Failure to attend shadow orientation hosted by EA (we host 2-3 sessions in different timezone)

Reviewer 2: Can we write out Emeritus Advisor here or mention the abbreviation in an earlier section?

- Failure to attend shadow orientation hosted by role lead (if lead doesn't offer timezone-appropriate meeting, shadow should communicate to lead about obtaining and watching the recording)

> - Failure to attend 3 consecutive release meeting without communication to role lead

Reviewer: ```suggestion
- Failure to attend 3 consecutive release meetings without communication to role lead
```

> - Failure to attend 3 consecutive release meeting without communication to role lead

Reviewer 2: So to clarify, missing consecutive release meetings is fine as long as it's been cleared by the role lead, correct? The different time zones make it a little tricky for some of the shadows to go to all of the release team meetings so it would be good to add this to the shadow application process. It looks like it's missing from the[ release shadows doc](https://github.com/kubernetes/sig-release/blob/master/release-team/shadows.md) and we could potentially add a point to the [logistics section](https://github.com/kubernetes/sig-release/blob/master/release-team/release-team-selection.md#logistics) in the release team selection as well.

- Failure to perform shadow tasks in due time, after 2 discussions with role lead about performance
- Unresponsive on Slack to messages for more than 2 days during non-peak time and 24 hours during peak time, without prior notice of absence. Peak time is different depending on the team's responsibility

> - Violation of CNCF Code of Conduct

Reviewer: ```suggestion
- Violation of [CNCF Code of Conduct](http://cncf.io/conduct/)
```

Note that communications ahead of time to inform the leads or find alternative arrangements are greatly preferred in these circumstances.
### Process

> It is important to be kind and patient while trying to resolve the situation.

Reviewer 2: I would add a bullet point to clarify this point `Failure to perform shadow tasks in due time, after 2 discussions with role lead about performance`. I don't think this would immediately trigger reaching out to the EA. I would first reach out to the shadow directly over slack to check in. Personally, if after 2 discussions of the shadow failing to perform tasks on time, then I would only raise it to the EA. 

```suggestion
### Process
It is important to be kind and patient while trying to resolve the situation.
- Notify the shadow in private message over slack of missed deadline and first warning.
- On second occurrence, schedule 1:1 meeting with shadow over slack and discuss performance.
```

> - Upon a pattern of inactivity, role lead will notify EA and have initial discussion with shadows about performance.

Reviewer: ```suggestion
- Upon a pattern of inactivity, the role lead will notify EA and have initial discussion with shadows about performance.
```

> - If the shadow is still not contributing to the team, EA and lead will remove shadow from team, and find replacement if appropriate. EA will be responsible to find backfill.

Reviewer: ```suggestion
- If the shadow is still not contributing to the team, EA and lead will remove the shadow from the team, and find a replacement if appropriate. EA will be responsible for finding backfill.
```

### Consequences
- If the current release is the reason the shadow has Kubernetes membership, membership should be revoked
- Removal of name in release-team document
- Shadow will not be able to return to the following release. However, they are welcome to apply to releases after that.

> # Role Lead

Reviewer: ```suggestion
## Role Lead Removal Guideline
```

A role lead has been selected by your previous team lead to supervise, perform the duties of your team 
and mentor your shadows to be future leads. Failure to perform lead duty can have a lasting impact on the release team.
Reasons for removal could be including but not limited to:
- Failure to attend or send appropriate representation to 2 consecutive release meetings
- Failure to perform tasks expected of a role lead
- Unresponsive on Slack to messages for more than 2 days during non-peak time and 24 hours during peak time, without prior notice of absence. Peak time is different depending on team

> - Violation of CNCF Code of Conduct

Reviewer: ```suggestion
- Violation of [CNCF Code of Conduct](http://cncf.io/conduct/)
```

### Process
- SIGs outside of SIG-release, shadows, leads, or contributors can express concern to release lead or EA.
- Release lead and EA will have initial discussion with role lead

> - If the situation is not improved, release lead and EA will have another discussion with role lead and begin to look for replacement.

Reviewer: ```suggestion
- If the situation is not improved, the release lead and EA will have another discussion with the role lead and begin to look for a replacement.
```

> - If deemed necessary by release lead, EA and sig-release chairs, role lead will be replaced with an experience shadow or someone who previously lead the same role. EA will be responsible to find backfill

Reviewer: ```suggestion
- If deemed necessary by the release lead, EA and sig-release chairs, the role lead will be replaced with an experience shadow or someone who previously led the same role. EA will be responsible for finding backfill
```

  
### Consequences

> - Removal of name in release-team document

Reviewer: ```suggestion
- Removal of name in the release-team document which documents the release team members of the release cycle
```

- Role lead will need to shadow again and demonstrate sufficient responsibility, and leading ability, before holding another leadership position on the release team

> # Appeal

Reviewer: ```suggestion
## Appeal
```

# Release Team Removal Process
	
The release team is reassembled with a new team for each cycle. All team members are automatically removed if they do not re-apply and are selected for the next release cycle. While this process generally functions smoothly, there are instances when a team member might leave mid-cycle or the release team leaders may decide to part ways early. 

## Resignation

> Voluntary resignation is outlined in the [shadow handbook](https://github.com/leonardpahlke/sig-release/blob/master/release-team/shadows.md). Role lead should discuss with Release Lead and Emeritus Advisor (EA) to find a replacement if they would like to step down.

Reviewer: ```suggestion
Voluntary resignation is outlined in the [shadow handbook](https://github.com/kubernetes/sig-release/blob/master/release-team/shadows.md#leaving-the-shadow-program-before-the-release-cycle-ended). The Role Lead should discuss with the Release Lead and the Emeritus Advisor (EA) to find a replacement if they would like to step down.
```

> Voluntary resignation is outlined in the [shadow handbook](https://github.com/leonardpahlke/sig-release/blob/master/release-team/shadows.md). Role lead should discuss with Release Lead and Emeritus Advisor (EA) to find a replacement if they would like to step down.

Reviewer 4: ```suggestion
Voluntary resignation is outlined in the [shadow handbook](https://github.com/leonardpahlke/sig-release/blob/master/release-team/shadows.md). The role lead should discuss with Release Lead and Emeritus Advisor (EA) to find a replacement if they would like to step down.
```

> Remaining on the team without contributing creates an unhealthy environment for the rest of the team and negatively affects the outcome of the release. We strive to be inclusive and empathetic to individuals' circumstances.

Reviewer 4: ```suggestion
Remaining on the team without contributing creates an unhealthy environment for the rest of the team and negatively affects the outcome of the release. We strive to be inclusive and empathetic to individuals' circumstances, but our first priority is a stable, high-quality release.
```

Adding a bit at the end here just because it feels a touch abrupt without.

Reviewer: 👍

The following outline is not meant to be definitive or punitive but rather outlines general guidance for leads on the process.  
## Shadow Removal Guideline

Reasons for removal could be including but not limited to:

> - Failure to attend shadow orientation hosted by the EA (we host 2-3 sessions in different timezones)

Reviewer: ```suggestion
- Failure to attend the shadow orientation hosted by the EA (we host 2-3 sessions at the beginning of the release cycle in different timezones)
```

> - Failure to attend shadow orientation hosted by the role lead (if lead doesn't offer timezone-appropriate meeting, shadow should communicate to lead about obtaining and watching the recording)

Reviewer: ```suggestion
- Failure to attend the orientation hosted by the Role Lead (if the Role Lead doesn't offer a timezone-appropriate meeting, the shadow should communicate to the Role Lead about obtaining the provided information async and watching the recording)
```

We have the shadow orientation by the EA. I would not name team sync’s shadow orientation meetings as it might confuse folks.

> - Failure to attend 3 consecutive release meetings without communication to the role lead

Reviewer: ```suggestion
- Failure to attend 3 consecutive release team meetings without communication to the Role Lead
```

> - Failure to perform shadow tasks in due time, after 2-3 discussions with the role lead and the EA about performance

Reviewer: ```suggestion
- Failure to perform shadow tasks in due time, after 2-3 discussions with the Role Lead and the EA about performance
```

- Unresponsive on Slack to messages for more than 2 days during non-peak time and 24 hours during peak time, without prior notice of absence. Peak time is different depending on the team's responsibility
- Violation of [CNCF Code of Conduct](http://cncf.io/conduct/)

Note that communications ahead of time to inform the leads or find alternative arrangements are greatly preferred in these circumstances.
### Process
It is important to be kind and patient while trying to resolve the situation.

> - Upon a pattern of inactivity, role lead will notify EA and have initial discussion with shadows about performance

Reviewer: ```suggestion
- Upon a pattern of inactivity, the Role Lead will notify the EA and have an initial discussion with the shadow about their performance.
```

> - If the shadow is still not contributing to the team, EA and lead will remove the shadow from the team, and find a replacement if appropriate. EA will be responsible for finding backfill.

Reviewer: ```suggestion
- If the shadow is still not contributing to the team, the EA and the Release Lead will remove the shadow from the team, and find a replacement if appropriate. The EA will be responsible for finding backfill.
```

### Consequences
- If the current release is the reason the shadow has Kubernetes membership, membership should be revoked
- Removal of name in release-team document
- Shadow will not be able to return to the following release. However, they are welcome to apply to releases after that

## Role Lead Removal Guideline

> A role lead has been selected by your previous team lead to supervise, perform the duties of your team 

Reviewer 4: ```suggestion
A role lead has been selected by the previous team lead to supervise, perform the duties of the team 
```

> and mentor your shadows to be future leads. Failure to perform lead duty can have a lasting impact on the release team.

Reviewer 4: ```suggestion
and mentor the shadows to be future leads. Failure to perform lead duties can have a lasting impact on the release team.
```

> A role lead has been selected by your previous team lead to supervise, perform the duties of your team 
> and mentor your shadows to be future leads. Failure to perform lead duty can have a lasting impact on the release team.

Reviewer: ```suggestion
A Role Lead has been selected to supervise a sub-team within the release team. This role encompasses not only overseeing team activities and contributing to the release cycle but also mentoring shadows throughout the cycle. The work of the Role Leads is pivotal in ensuring a successful release cycle, while simultaneously mentoring shadows to develop them into new contributors within the Kubernetes community. Failure to perform the lead duty can have a lasting impact on the release team.
```

rephrased, ptal.

> Reasons for removal could be including but not limited to:

Reviewer 4: ```suggestion
Reasons for removal could include but are not limited to:
```

> - Failure to attend or send appropriate representation to 2 consecutive release meetings

Reviewer 4: ```suggestion
- Failure to attend or send appropriate representation to 2 consecutive release meetings or relevant SIG meetings
```

- Failure to perform tasks expected of a role lead
- Unresponsive on Slack to messages for more than 2 days during non-peak time and 24 hours during peak time, without prior notice of absence. Peak time is different depending on team
- Violation of [CNCF Code of Conduct](http://cncf.io/conduct/)

### Process

> - SIGs outside of SIG-release, shadows, leads, or contributors can express concern to release lead or EA.

Reviewer 4: ```suggestion
- SIGs outside of SIG Release, shadows, leads, or contributors can express concern to release lead or EA.
```

> - Release lead and EA will have initial discussion with role lead

Reviewer: ```suggestion
- The Release Lead and the EA will have an initial discussion with the Role Lead
```

> - If the situation is not improved, the release lead and EA will have another discussion with role lead and begin to look for a replacement.

Reviewer: ```suggestion
- If the situation does not improve within a week after the first discussion, the Release Lead and the EA will have another discussion with the Role Lead and begin to look for a replacement.
```

within a week?

Author: it might take more than a week to get signal. this is also more general guideline

> - If deemed necessary by the release lead, EA and sig-release chairs, the role lead will be replaced with an experience shadow or someone who previously led the same role. EA will be responsible for finding backfill

Reviewer 4: ```suggestion
- If deemed necessary by the release lead, EA and SIG Release chairs, the role lead will be replaced with an experience shadow or someone who previously led the same role. EA will be responsible for finding backfill
```

  
### Consequences
- Removal of name in release-team document
- Role lead will need to shadow again and demonstrate sufficient responsibility, and leading ability, before holding another leadership position on the release team

## Appeal

> If the release team member disagrees with their removal decision. They can escalate to SIG-Release chairs via private messaging on Slack.

Reviewer 4: ```suggestion
If the release team member disagrees with their removal decision, they can escalate to SIG Release chairs via private messaging on Slack or by emailing sig-release-leads@kubernetes.io.
```

Maybe emailing all of the SIG Release leads at once is better here? Versus potentially DMing only one of them.

Author: removed the DM option
